group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -669,7 +668,12 @@ private void throwIfConsumerGroupHeartbeatRequestIsInvalid(
                 throw new InvalidRequestException("TopicPartitions must be empty when (re-)joining.");
             }
             if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
-                throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
+                if (request.subscribedTopicRegex() == null || request.subscribedTopicRegex().isEmpty()) {
+                    throw new InvalidRequestException("SubscribedTopicNames or SubscribedTopicRegex must be set in first request.");
+                }
+            }
+            if (request.subscribedTopicRegex() != null && !request.subscribedTopicNames().isEmpty()  && !isEmpty(request.subscribedTopicRegex())) {
+                throw new InvalidRequestException("SubscribedTopicNames or SubscribedTopicRegex should not be set at the same time.");
```

Comment:
nit: `Both SubscribedTopicNames and SubscribedTopicRegex should not be set at the same time.`